sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -333,6 +334,107 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
     return true;
   }
 
+  /**
+   * Updates the last reported assessment for the current assessment.
+   * @param assessmentId related assessment id.
+   * @param entityManager entity manager to be part of the transaction.
+   * @returns updated assessment.
+   */
+  async updateLastReportedAssessment(
+    assessmentId: number,
+    entityManager: EntityManager,
+  ): Promise<UpdateResult | undefined> {
+    const assessment = await this.repo.findOne({
+      select: {
+        id: true,
+        offering: {
+          studyStartDate: true,
+          studyEndDate: true,
+        },
+        application: {
+          id: true,
+        },
+      },
+      relations: {
+        application: true,
+        offering: true,
+      },
+      where: { id: assessmentId },
+    });
+    const studentAssessmentRepo =
+      entityManager.getRepository(StudentAssessment);
+    const auditUser = this.systemUsersService.systemUser;
+    const lastReportedAssessment = await this.getLastReportedAssessment(
+      assessment.application.id,
+      entityManager,
+    );
+    if (
+      !lastReportedAssessment ||
+      (assessment.offering.studyStartDate ===
+        lastReportedAssessment.offering.studyStartDate &&
+        assessment.offering.studyEndDate ===
+          lastReportedAssessment.offering.studyEndDate)
+    ) {
+      return;
+    }
+    return studentAssessmentRepo.update(
+      {
+        id: assessmentId,
+      },
+      {
+        previousDateChangedReportedAssessment: lastReportedAssessment,
+        modifier: auditUser,
+      },
+    );
+  }
+
+  /**
+   * Gets the last reported assessment.
+   * @param applicationId current assessment application id.
+   * @param entityManager entity manager to be part of the transaction.
+   * @returns last reported assessment.
+   */
+  private async getLastReportedAssessment(
+    applicationId: number,
+    entityManager: EntityManager,
+  ): Promise<StudentAssessment> {
+    const studentAssessmentRepo =
+      entityManager.getRepository(StudentAssessment);
+    const lastApplicationChangeReportedAssessment =
+      await studentAssessmentRepo.findOne({
+        select: {
+          id: true,
+          assessmentDate: true,
+          offering: { id: true, studyStartDate: true, studyEndDate: true },
+        },
+        relations: { offering: true },
+        where: {
+          application: { id: applicationId },
+          previousDateChangedReportedAssessment: Not(IsNull()),
+          reportedDate: Not(IsNull()),
+        },
+        order: { assessmentDate: "DESC" },
+      });
+    if (lastApplicationChangeReportedAssessment) {
+      return lastApplicationChangeReportedAssessment;
+    }
+    return studentAssessmentRepo.findOne({
+      select: {
+        id: true,
+        assessmentDate: true,
```

Comment:
same here on assessment date.

Reply:
Yes, it is required for the Order. Without it, being selected, typeorm throws error.